pkg/microservice/warpdrive/core/service/taskplugin/job.go

```diff
@@ -631,67 +633,66 @@ func getVolumes(jobName string) []corev1.Volume {
-			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("16"),
-				corev1.ResourceMemory: resource.MustParse("32Gi"),
-			},
-			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("4"),
-				corev1.ResourceMemory: resource.MustParse("4Gi"),
-			},
-		}
+		return generateResourceRequirements(setting.HighRequest, setting.HighRequestSpec)
+
 	case setting.MediumRequest:
-		return corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("8"),
-				corev1.ResourceMemory: resource.MustParse("16Gi"),
-			},
-			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("2"),
-				corev1.ResourceMemory: resource.MustParse("2Gi"),
-			},
-		}
+		return generateResourceRequirements(setting.MediumRequest, setting.MediumRequestSpec)
+
 	case setting.LowRequest:
-		return corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("4"),
-				corev1.ResourceMemory: resource.MustParse("8Gi"),
-			},
-			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("1"),
-				corev1.ResourceMemory: resource.MustParse("1Gi"),
-			},
-		}
+		return generateResourceRequirements(setting.LowRequest, setting.LowRequestSpec)
 
 	case setting.MinRequest:
-		return corev1.ResourceRequirements{
-			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("2"),
-				corev1.ResourceMemory: resource.MustParse("2Gi"),
-			},
-			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("0.5"),
-				corev1.ResourceMemory: resource.MustParse("512Mi"),
-			},
-		}
+		return generateResourceRequirements(setting.MinRequest, setting.MinRequestSpec)
+
+	case setting.DefineRequest:
+		return generateResourceRequirements(resReq, resReqSpec)
+
 	default:
+		return generateResourceRequirements(setting.DefaultRequest, setting.DefaultRequestSpec)
+	}
+}
+
+//generateResourceRequirements
+//cpu Request:Limit=1:4
+//memory default Request:Limit=1:4 ; if memoryLimit>8Gi,Request:Limit=1:8
+func generateResourceRequirements(req setting.Request, reqSpec setting.RequestSpec) corev1.ResourceRequirements {
+
+	if req != setting.DefineRequest {
 		return corev1.ResourceRequirements{
 			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("4"),
-				corev1.ResourceMemory: resource.MustParse("8Gi"),
+				corev1.ResourceCPU:    resource.MustParse(reqSpec.CpuLimit),
+				corev1.ResourceMemory: resource.MustParse(reqSpec.MemoryLimit),
 			},
 			Requests: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("1"),
-				corev1.ResourceMemory: resource.MustParse("1Gi"),
+				corev1.ResourceCPU:    resource.MustParse(reqSpec.CpuReq),
+				corev1.ResourceMemory: resource.MustParse(reqSpec.MemoryReq),
 			},
 		}
 	}
+
+	// DefineRequest
+	cpuLimitInt, _ := strconv.Atoi(strings.TrimSuffix(reqSpec.CpuLimit, setting.CpuUintM))
```

Comment:
Unuseful comments should be removed.

We can consider using [https://github.com/kubernetes/apimachinery/tree/master/pkg/api/resource](https://github.com/kubernetes/apimachinery/tree/master/pkg/api/resource) later.